File: matlab/ex01_randomwalk.m
%% Random Walk

% Select a desired Theme
hebi_charts.Runtime.setTheme('NordDark');

% Create a 1x1 grid window with a single chart element
window = hebi_charts.GridWindow('title', 'Python Random Walk', 'size', [800 600]);

% Add a line chart element to the window
chart = window.addLineChart( ...
    'title', 'Random Walk', ...
    'xlabel', 'sample', ...
    'ylabel', 'rnd(x)');

% Add a trace with solid line style and max point limit
line = chart.addLine('walk', ...
    'lineStyle', 'Solid', ...
    'maxPointCount', 10000);

%% Update loop
x = 0;
y = 0;
window.show();
while window.isShowing()
    % Reduce the rate (optional)
    pause(0.001);

    % Append one point (non-blocking)
    x = x + 1;
    y = y + (rand() * 2 - 1);
    line.addPoint(x, y);
end

File: python/ex01_randomwalk.py
import time
import random
import hebi_charts

def main():
    # Select a desired Theme
    hebi_charts.Runtime.set_theme(hebi_charts.Theme.NORD_DARK)

    # Create a 1x1 grid window with a single chart element
    window = hebi_charts.GridWindow(title="Python Random Walk", size=(800, 600))
    chart = window.add_line_chart(title="Random Walk", xlabel="sample", ylabel="rnd(x)")

    # Add a trace
    line = chart.add_line("walk", line_style=hebi_charts.LineStyle.SOLID, max_point_count=10_000)

    # Continuously update a random walk
    x, y = 0, 0
    window.show()
    while window.is_showing():
        # reduce the rate (optional)
        time.sleep(0.001)

        # append one point (non-blocking)
        x += 1
        y += random.uniform(-1, 1)
        line.add_point(x, y)

# macOS needs to own the main thread, so we need to
# do a handover. no-op on other operating systems.
if __name__ == "__main__":
    hebi_charts.run_application(main)
